substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/AbstractDynamicClassGenerationSupport.java

```diff
@@ -0,0 +1,196 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.agent;
+
+import com.oracle.svm.core.util.JavaClassUtil;
+import static com.oracle.svm.jvmtiagentbase.Support.jniFunctions;
+import static com.oracle.svm.jvmtiagentbase.Support.getMethodFullNameAtFrame;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import com.oracle.svm.jni.nativeapi.JNIMethodId;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+import org.graalvm.nativeimage.c.type.CTypeConversion;
+import org.graalvm.nativeimage.c.type.VoidPointer;
+import org.graalvm.word.WordFactory;
+
+import com.oracle.svm.jni.nativeapi.JNIEnvironment;
+import com.oracle.svm.jni.nativeapi.JNIObjectHandle;
+
+public abstract class AbstractDynamicClassGenerationSupport {
+
+    protected JNIEnvironment jni;
+    protected JNIObjectHandle callerClass;
+    protected final String generatedClassName;
+    protected TraceWriter traceWriter;
+    protected NativeImageAgent agent;
+    protected int generatedClassHashCode = 0;
+    protected byte[] values = null;
+
+    private static Path dynclassDumpDir = null;
+    private static final Path DEFAULT_DUMP = Paths.get("dynClass");
+    protected String callerMethod;
+
+    public static void initDynClassDumpDir(String dir) {
+        dynclassDumpDir = Paths.get(dir);
+    }
+
+    public static AbstractDynamicClassGenerationSupport getClassLoaderDefineClassSupport(JNIEnvironment jni, JNIObjectHandle callerClass,
+                    String generatedClassName, TraceWriter traceWriter, NativeImageAgent agent) throws IOException {
+        return new ClassLoaderDefineClassSupport(jni, callerClass, generatedClassName, traceWriter, agent);
+    }
+
+    protected AbstractDynamicClassGenerationSupport(JNIEnvironment jni, JNIObjectHandle callerClass,
+                    String generatedClassName, TraceWriter traceWriter, NativeImageAgent agent) throws IOException {
+        if (dynclassDumpDir == null) {
+            System.out.println("Warning: dynamic-class-dump-dir= was not set in -agentlib:native-image-agent=. Dynamically generated classes will be dumped to the default location:" +
+                            DEFAULT_DUMP.toAbsolutePath().toString());
+            dynclassDumpDir = DEFAULT_DUMP;
+        }
+
+        if (!Files.exists(dynclassDumpDir)) {
+            Files.createDirectories(dynclassDumpDir);
+        } else if (!Files.isDirectory(dynclassDumpDir)) {
+            throw new IOException("File " + dynclassDumpDir + " already exists! Cannot create the same directory for class file dumping.");
```

Comment:
I would mention here that the file in place is a file and that we want to create a dir there.

Reply:
fixed